src/server/main_service.cc

```diff
@@ -878,10 +878,13 @@ optional<ErrorReply> Service::CheckKeysOwnership(const CommandId* cid, CmdArgLis
     return ErrorReply{kClusterNotConfigured};
   }
 
-  if (keys_slot.has_value() && !cluster_config->IsMySlot(*keys_slot)) {
-    // See more details here: https://redis.io/docs/reference/cluster-spec/#moved-redirection
-    ClusterConfig::Node master = cluster_config->GetMasterNodeForSlot(*keys_slot);
-    return ErrorReply{absl::StrCat("-MOVED ", *keys_slot, " ", master.ip, ":", master.port)};
+  if (keys_slot.has_value()) {
+    ServerState::tlocal()->AwaitIfMigrationFinalization(*keys_slot);
```

Comment:
this is an unfortunate place to block, verify function should be immediate... Can we block somehow from DispatchCommand?

Reply:
TODO(myself): Add some comments or DCHECKs about this

The reason for is that Verify commands are called when either collecting multi commands or preparing squashing  and sometimes they're not on the same code path as the executed command

Reply:
Actually it's very hard now 😅 😅 😅 

Scenario:
```
MULTI
SET a 1 -> ok stored
SET b 1 -> ok stored
SET C 1 -> ok stored 
-- finalize-slot-migration arrives --
EXEC -> we need to check a b c here, not above
```

Will come up with ideas tomorrow

Reply:
I suggest using basic client pause for now. We have enough problems in the existing migration code, so why should we strive for a perfect solution here. To reduce the blocking time we can lower the replication gap ahead